src/PubSubClient.cpp

```diff
@@ -524,6 +524,7 @@ void PubSubClient::disconnect() {
     buffer[1] = 0;
     _client->write(buffer,2);
```

Comment:
I noticed the `_client` pointer is never checked before dereferencing.
This makes it tricky to call the disconnect before switching off the wifi or go back to sleep, since you have to keep track outside of this library if the client ever has been set.